gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemDelegationTokensTest.java

```diff
@@ -139,6 +140,89 @@ public void testDelegationTokenStatistics() throws IOException {
     fs.close();
   }
 
+  @Test
+  public void testDelegationTokenStatisticsForSharedTokens() throws IOException {
+
+    GoogleHadoopFileSystem fs = new GoogleHadoopFileSystem();
+    fs.initialize(new Path("gs://test/").toUri(), loadSharedDelegationTokenConfig());
+
+    Token<?> dt = fs.getDelegationToken("shared_token_test1");
+    assertThat(fs.getIOStatistics().counters().get(INVOCATION_GET_DELEGATION_TOKEN.getSymbol()))
+        .isEqualTo(1);
+    assertThat(fs.getIOStatistics().counters().get(DELEGATION_TOKENS_ISSUED.getSymbol()))
+        .isEqualTo(1);
+
+    // second test for renewer which is null
+    Token<?> dt2 = fs.getDelegationToken(null);
+    assertThat(fs.getIOStatistics().counters().get(INVOCATION_GET_DELEGATION_TOKEN.getSymbol()))
+        .isEqualTo(2);
+    assertThat(fs.getIOStatistics().counters().get(DELEGATION_TOKENS_ISSUED.getSymbol()))
+        .isEqualTo(2);
+    fs.close();
+  }
+
+  @Test
+  public void testDelegationTokensAreSharedForSharedTokensConfigAndSameRenewers()
+      throws IOException {
+
+    GoogleHadoopFileSystem fs = new GoogleHadoopFileSystem();
+    fs.initialize(new Path("gs://test/").toUri(), loadSharedDelegationTokenConfig());
+
+    GoogleHadoopFileSystem fsOther = new GoogleHadoopFileSystem();
+    fsOther.initialize(new Path("gs://test2/").toUri(), loadSharedDelegationTokenConfig());
+
+    Token<?> dt1 = fs.getDelegationToken("current_user");
+    Token<?> dt2 = fsOther.getDelegationToken("current_user");
+    assertThat(dt1).isEqualTo(dt2);
```

Comment:
`Token` implements a value `equals`, so in theory, there could be a bug introduced that produces 2 separate but equivalent instances. I suggest changing from `isEqualTo` to `isSameInstanceAs`.

Reply:
done